File: presentation/src/main/java/cleanarcpro/brightowusu/com/cleanarcproj/di/modules/RepositoryModule.java
package cleanarcpro.brightowusu.com.cleanarcproj.di.modules;

import android.content.Context;

import javax.inject.Singleton;

import cleanarcpro.brightowusu.com.cleanarcproj.data.repository.UserRepositoryImpl;
import cleanarcpro.brightowusu.com.cleanarcproj.domain.abstractions.repository.IUserRepository;
import dagger.Module;
import dagger.Provides;

@Module//(includes = AppModule.class)
public class RepositoryModule {

    @Singleton
    @Provides
    public IUserRepository providesUserRepository() {
        return new UserRepositoryImpl();
    }

}

File: presentation/src/main/java/cleanarcpro/brightowusu/com/cleanarcproj/di/modules/RepositoryModule.kt
package cleanarcpro.brightowusu.com.cleanarcproj.di.modules;

import cleanarcpro.brightowusu.com.cleanarcproj.data.repository.db.dao.PastExperienceDao
import cleanarcpro.brightowusu.com.cleanarcproj.data.repository.db.dao.ProSummaryDao
import cleanarcpro.brightowusu.com.cleanarcproj.data.repository.db.dao.UserDao
import javax.inject.Singleton

import cleanarcpro.brightowusu.com.cleanarcproj.data.repository.utils.RepositoryProvider
import cleanarcpro.brightowusu.com.cleanarcproj.domain.abstractions.repository.IUserRepository;
import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit

/**
 * Repository Implementations.
 *
 * Created by Bright Owusu-Amankwaa
 */
@Module(includes = [AppModule::class])
class RepositoryModule {

    @Singleton
    @Provides
    fun providesUserRepository(
            retrofit: Retrofit,
            userDao: UserDao,
            pastExperienceDao: PastExperienceDao,
            proSummaryDao: ProSummaryDao
            ): IUserRepository {
        return RepositoryProvider.providesUserRepository(
                retrofit,
                userDao,
                proSummaryDao,
                pastExperienceDao)
    }

}
